src/main/java/org/generationcp/middleware/pojos/workbench/WorkbenchUser.java

```diff
@@ -66,21 +65,89 @@
+		+ "   where  (r.role_type_id =  " + RoleType.INSTANCE.getId()
+		+ "     or (r.role_type_id = " + RoleType.CROP.getId() + " and ur.crop_name = :cropName)  "
+		+ "     or (r.role_type_id =  " + RoleType.PROGRAM.getId()
+		+ " and ur.crop_name = :cropName AND ur.workbench_project_id = p.project_id))  "
+		+ "    AND "
+		+ "       p.project_id = :projectId "
+		+ "    GROUP BY users.userid";
+
+	public static final String GET_ACTIVE_USER_IDS_BY_PROJECT_ID =
+		"SELECT DISTINCT users.userid "
+			+ "    FROM "
+			+ "       workbench_project p "
+			+ "           INNER JOIN "
+			+ "       workbench_crop wc ON p.crop_type = wc.crop_name "
+			+ "           INNER JOIN "
+			+ "       crop_persons cp ON cp.crop_name = wc.crop_name "
+			+ "           INNER JOIN "
+			+ "       users ON cp.personid = users.personid "
+			+ "           INNER JOIN "
+			+ "       users_roles ur ON ur.userid = users.userid "
+			+ "           INNER JOIN role r ON ur.role_id = r.id  "
+			+ "   where  (r.role_type_id =  " + RoleType.PROGRAM.getId()
+			+ " 			AND ur.crop_name = :cropName AND ur.workbench_project_id = p.project_id) "
+			+ "    	AND p.project_id = :projectId "
+			+ "  	AND users.ustatus = 0 ";
+
 	public static final String GET_BY_NAME_USING_EQUAL = "getUserByNameUsingEqual";
 	public static final String GET_BY_NAME_USING_LIKE = "getUserByNameUsingLike";
 	public static final String GET_ALL_ACTIVE_USERS_SORTED = "getAllActiveUsersSorted";
 	public static final String GET_BY_FULLNAME = "getByFullName";
 
-	public static final String GET_USERS_BY_PROJECT_UUID =
-		"SELECT users.userid, users.uname, person.fname, person.lname, role.id, role.description, users.ustatus, person.pemail \n"
-		+ "FROM users \n"
-		+ "INNER JOIN workbench_project_user_info pu ON users.userid = pu.user_id \n"
-		+ "INNER JOIN persons person ON person.personid = users.personid \n "
-		+ "INNER JOIN workbench_project pp ON pu.project_id = pp.project_id \n "
-		+ "INNER JOIN users_roles ur ON ur.userid = users.userid "
-		+ "INNER JOIN role ON role.id = ur.role_id "
-		+ "WHERE pp.project_uuid = :project_uuid \n "
-		+ "GROUP BY users.userid";
+	public static final String GET_USERS_BY_PROJECT_UUID = "SELECT  "
+		+ "       u.userid, "
+		+ "       u.uname, "
+		+ "       person.fname, "
+		+ "       person.lname, "
+		+ "       u.ustatus, "
+		+ "       person.pemail "
+		+ "    FROM "
+		+ "       workbench_project p "
+		+ "           INNER JOIN "
+		+ "       workbench_crop wc ON p.crop_type = wc.crop_name "
+		+ "           INNER JOIN "
+		+ "       crop_persons cp ON cp.crop_name = wc.crop_name "
+		+ "           INNER JOIN "
+		+ "       persons person ON person.personid = cp.personid "
+		+ "           INNER JOIN "
+		+ "       users u ON person.personid = u.personid "
+		+ "           INNER JOIN "
+		+ "       users_roles ur ON ur.userid = u.userid "
+		+ "           INNER JOIN role r ON ur.role_id = r.id  "
+		+ "   where  (r.role_type_id =  " + RoleType.INSTANCE.getId()
```

Comment:
The criteria below is duplicated in several queries. Better to extract to separate constant:
```
(r.role_type_id =  " + RoleType.INSTANCE.getId()
		+ "     or (r.role_type_id = " + RoleType.CROP.getId() + " and ur.crop_name = :cropName)  "
		+ "     or (r.role_type_id =  " + RoleType.PROGRAM.getId() +" and ur.crop_name = :cropName AND ur.workbench_project_id = p.project_id))
```